src/libraries/Microsoft.Extensions.HostFactoryResolver/src/HostFactoryResolver.cs

```diff
@@ -31,6 +37,16 @@ internal sealed class HostFactoryResolver
             return ResolveFactory<THostBuilder>(assembly, CreateHostBuilder);
         }
 
+        public static Func<string[], IHostBuilder>? ResolveHostBuilderFactory(Assembly assembly)
```

Comment:
1. Does this need to be public?
2. If it does, it is a bit concerning that it has the same name as the method above, takes the same parameters, and returns the same type, but doesn't really do the same things. Can we rename it somehow to distinguish it from the existing method above?

Reply:
I'll rename it.